### PR TITLE
Adds default values to MicroResourceVariable

### PR DESCRIPTION
**Pull request by suleshahid**
Currently we memset MicroResourceVariables buffers to 0 if it is null, and in reset we memset to 0 again.
This change makes it so that we memset to the zero_point instead.

To get the information of the zero_point value while reset, we need to store it in the MicroResourceVariable class.

BUG=[269648474](https://b.corp.google.com/issues/269648474)

### CODE REVIEW

File: tensorflow/lite/micro/micro_resource_variable.cc [hunk 1]
```diff
@@ -96,9 +97,14 @@ TfLiteStatus MicroResourceVariables::Allocate(int id, TfLiteContext* context,
       MicroPrintf("Failed to allocate resource buffer.");
       return kTfLiteError;
     }
-    // Zero out resource buffers by deafult. Buffers can be initialized to
-    // nonzero values using ASSIGN_VARIABLE.
-    memset(variable.resource_buffer, 0, variable.bytes);
+    // Set resource buffers to the zero_point by default. Buffers can be
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_resource_variable.cc:100]: let's add a comment here that link to a specific comment within the bug for more context. That will help the reader understand why this was done. And then we can figure out if zero point is the generic solution or if we want to go in a different direction later.

**Review by advaitjain — CHANGES_REQUESTED** [review]
what is MRV in the PR description? can you update it to the full text?

File: tensorflow/lite/micro/micro_resource_variable.cc (hunk 1 above)
Comment (suleshahid) [reply tensorflow/lite/micro/micro_resource_variable.cc:100]: Done.

**Review by advaitjain — DISMISSED** [review]

**Comment by suleshahid** [comment]
Note, not going to merge this yet. I have cl/510282473 for testing this in g3 as well and we are probably going to need to change a lot of arena sizes if they were set exact and use resource variables.

**Review by paulinesho — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

**Pull request merged**